data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/AbstractCascadeOperations.java

```diff
@@ -185,7 +189,7 @@ protected <T> T afterCascadedMany(T entity, List<Association> associations, Iter
                     }
                 }
             }
-            if (prevChildren != newChildren) {
+            if (association.getProperty().get(entity) != newChildren) {
```

Comment:
Another fix for cascade save, without this when entity saved then cascade items are not set property after save.